File: backend/src/main/java/com/demo/travel_expense_management/refund/service/RefundService.java
package com.demo.travel_expense_management.refund.service;

import com.demo.travel_expense_management.refund.dto.request.RegisterRefundRequest;
import com.demo.travel_expense_management.refund.dto.response.RefundResponse;

public interface RefundService {

    RefundResponse registerRefund(Long legalizationId, RegisterRefundRequest request);
}

File: frontend/src/lib/services/refund.service.ts
import { apiClient } from '$lib/api/api-client';
import type {
	RefundResponseDto,
	RegisterRefundDto,
} from '$lib/dto/refund';

class RefundService {
	private readonly resourcePath = '/v1/refunds';

	registerRefund(
		legalizationId: number,
		payload: RegisterRefundDto,
	): Promise<RefundResponseDto> {
		return apiClient.post<RefundResponseDto>(
			`${this.resourcePath}/legalization/${legalizationId}`,
			payload,
		);
	}
}

export const refundService = new RefundService();
